pytorch_lightning/strategies/ddp.py

```diff
@@ -152,6 +154,17 @@ def setup(self, trainer: "pl.Trainer") -> None:
         if trainer_fn == TrainerFn.FITTING:
             self.configure_ddp()
 
+        # set up optimizers after the wrapped module has been moved to the device
+        self.setup_optimizers(trainer)
```

Comment:
by moving the `setup_optimizers()` after the configure_ddp(), it will now affect the BaguaStrategy. 
Bagua runs checks on the optimizer in configuer_ddp:

```py
        if self._bagua_algorithm == "qadam":
            self._check_qadam_optimizer()
```

We would need to move it also after the setup_optimizer() call was made.

Reply:
great catch @awaelchli ! i forgot that bagua also extending DDP :/